### PR TITLE
DOC: update the pandas.DataFrame.plot.box docstring

### PR DESCRIPTION
**Pull request by 4kxz**
A bit late to the party, but I did it.

- [ ] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [ ] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

```
################################################################################
#################### Docstring (pandas.DataFrame.plot.box)  ####################
################################################################################

Make a box plot of the DataFrame columns.

A box plot is a method for graphically depicting
groups of numerical data through their quartiles.
The box extends from the Q1 to Q3 quartile values of the data,
with a line at the median (Q2). The whiskers extend from the edges
of box to show the range of the data. The position of the whiskers
is set by default to 1.5*IQR (IQR = Q3 - Q1) from the edges of the
box. Outlier points are those past the end of the whiskers.

For further details see Wikipedia's
entry for `boxplot <https://en.wikipedia.org/wiki/Box_plot>`_.

A consideration when using this chart is that the box and the whiskers
can overlap, which is very common when plotting small sets of data.

Parameters
----------
by : string or sequence
    Column in the DataFrame to group by.
**kwds : optional
    Additional keywords are documented in
    :meth:`pandas.DataFrame.plot`.

Returns
-------
axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them

See Also
--------
pandas.DataFrame.boxplot: Another method to draw a box plot.
pandas.Series.plot.box: Draw a box plot from a Series object.
matplotlib.pyplot.boxplot: Draw a box plot in matplotlib.

Examples
--------
Draw a box plot from a DataFrame with four columns of randomly
generated data.

.. plot::
    :context: close-figs

    >>> data = np.random.randn(25, 4)
    >>> df = pd.DataFrame(data, columns=list('ABCD'))
    >>> plot = df.plot.box()

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameters {'kwds'} not documented
		Unknown parameters {'**kwds'}
```

Due to issue #15079 I wasn't sure how to document the `by` argument. I will follow up to add it later.

Changed kwds description according to #20348.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -3031,19 +3031,46 @@ def barh(self, x=None, y=None, **kwds):
 
     def box(self, by=None, **kwds):
         r"""
-        Boxplot
+        Make a box plot of the DataFrame columns.
+
+        This method draws a box and whisker for each series in the DataFrame
+        in a single chart. The boxes share the vertical axis, which
+        makes makes it easier to compare if they are similar in scale.
```
Comment (dukebody) [inline pandas/plotting/_core.py]: "makes makes" → "makes"

**Review by dukebody — APPROVED** [review]
Apart from the typo in the extended description, LGTM!

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -3031,19 +3031,46 @@ def barh(self, x=None, y=None, **kwds):
...
+
+        Examples
+        --------
+        Draw a box plot from a DataFrame with four columns of randomly
+        generated data.
+
+        .. plot::
+            :context: close-figs
+
+            >>> data = np.random.randint(0, 10, size=(25, 4))
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: I think it is better to use normal distribution here (`np.random.randn(25, 4)`), it gives a bit more natural boxplots (eg with outliers) than a uniform distribution.

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -3031,19 +3031,46 @@ def barh(self, x=None, y=None, **kwds):
...
 
         Returns
         -------
         axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+
+        See Also
+        --------
+        :meth:`pandas.DataFrame.boxplot` : Another method to draw a box plot.
+        :meth:`pandas.Series.plot.box` : Draw a box plot from a Series object.
+        :func:`matplotlib.pyplot.boxplot`: Draw a box plot in matplotlib.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: The ``:meth:`.. `` part is not needed here, that is done automatically by sphinx.

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -3031,19 +3031,46 @@ def barh(self, x=None, y=None, **kwds):
...
     def box(self, by=None, **kwds):
         r"""
-        Boxplot
+        Make a box plot of the DataFrame columns.
+
+        This method draws a box and whisker for each series in the DataFrame
+        in a single chart. The boxes share the vertical axis, which
+        makes it easier to compare if they are similar in scale.
+        Box plots are useful because they clearly show the median,
+        quartiles and flier points in the data.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: There is also the PR on the `boxplot` method: https://github.com/pandas-dev/pandas/pull/20152. It has also extended the summary, you can check if there are some additions over there you could also use here (eg I think the wikipedia link is interesting) 

(ideally we would re-use this extended summary for both, but that is something we can fix later)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added some comments

File: pandas/plotting/_core.py (hunk 4 above)
Comment (Lkxz) [reply pandas/plotting/_core.py]: I've tried to fix the issues that you commented.

The summary from #20152 is more explanatory so I've copied most of it here. I removed the mention to the `by` parameter which, unless I'm missing something, is not yet implemented for plot.box. Is someone working on that?

**Review by dukebody — APPROVED** [review]
IMO this is good enough. We can consider creating some shared doc parts for both `DataFrame.plot.box` and `DataFrame.boxplot` later.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@Lkxz Thanks for the updates, and for the PR!

**Pull request merged**